### DISCUSSION TITLE
Deleting Identity Source

### DISCUSSION THREAD

**Discussion by ghost** [vmware/govmomi#3129] [Community Help] [answered]
Hi there!
I'm successfuly creating and updating AD over LDAP server using RegisterLdap and UpdateLdap methods from ssoadmin/client.go
There is no corresponding "unregister" method though and I see in PowerCLI example that "DeleteDomain" method is used.

I'm not able to call it though, as I'm getting

` ServerFaultCode: Invalid managed method 'DeleteDomain' requested for Managed Object Type 'SsoAdminIdentitySourceManagementService' version 'sso/version2'`

What am I doing wrong and how to implement removing identity sources?

Best regards

**Comment by dougm** [comment, marked as answer]
Hi @grzegorzgoldapl ,

The ssoadmin endpoint is tricky since there's no generated docs as there are for other API endpoints. I'm not sure if ssoadmin.wsdl is even included within vCenter anymore.  There are several related service IDs: https://github.com/vmware/govmomi/blob/88aa85c577ecbe7047df3f7ceb58ffabca4e4d20/ssoadmin/types/types.go#L381-L399

Looks like `DeleteDomain` is a `DomainManagementService` method, so the following should work:

```diff
modified   ssoadmin/client.go
@@ -571,3 +571,13 @@ func (c *Client) UpdateLdapAuthnType(ctx context.Context, name string, auth type
 	_, err := methods.UpdateLdapAuthnType(ctx, c, &req)
 	return err
 }
+
+func (c *Client) DeleteDomain(ctx context.Context, name string) error {
+	req := types.DeleteDomain{
+		This: c.ServiceContent.DomainManagementService,
+		Name: name,
+	}
+
+	_, err := methods.DeleteDomain(ctx, c, &req)
+	return err
+}
```

**Reply by ghost** [reply]
Thank you Doug!
